### ISSUE TITLE
local_hero for GridView items

### ISSUE THREAD

**Issue by jimmy-robert** [letsar/local_hero#1]
Hello @letsar ! 

Great package as always 👍🏾 I really think this can be a game changer for many use cases. 

I can't make it run with GridView items though, do you have any idea why?

Greetings from an former colleague from OAB 🙂

**Comment by letsar** [comment]
Hi Jimmy 🙂.
It's probably because of Repaint boundary. Try to set addRepaintBoundaries to false. I will look further later if it's not that.

**Comment by letsar** [comment]
Ok, so that's because the RepaintBoundary as I said, but if your GridView is scrollable, it will not work as you expect. During the scroll, all items will change positions, so they will try to animate, resulting in an ugly user experience 😕.

**Comment by jimmy-robert** [comment]
Indeed, I wish it worked with ListViews and GridViews, it would have been a great alternative to AnimatedList and (not existing) AnimatedGrid.

Great job though, other than in scrollable views it works like a charm. Already ⭐️ the repo.

**Comment by letsar** [comment]
Yeah I didn't designed it to work under scrollable widgets while it could be quite interesting. I don't know how to handle that for the moment but maybe there is a solution.
Thanks for the star :wink:

**Comment by rbozan** [comment]
Unfortunately that it doesn't work with ListViews and GridViews. I have an user interface where you can switch between a card layout and a list layout, and while the size does transition correctly the position is immediately changed.

Guess I'll just use a Navigator, as that also works for my use case :)

**Comment by lukepighetti** [comment]
Stick it in a custom TableGrid and it will work for you.

```dart
import 'package:flutter/material.dart';

class TableGrid extends StatelessWidget {
  /// A very simple grid without using scrollable solutions like [GridView]
  const TableGrid({
    Key key,
    this.columns = 2,
    @required this.children,
  }) : super(key: key);

  /// The number of columns to have in the width
  final int columns;

  final List<Widget> children;

  @override
  Widget build(BuildContext context) {
    return LayoutBuilder(
      builder: (context, constraints) {
        assert(constraints.hasBoundedWidth);
        final width = constraints.maxWidth;

        return Wrap(
          children: [
            for (var child in children)
              SizedBox(
                width: width / columns,
                child: child,
              ),
          ],
        );
      },
    );
  }
}
```

**Comment by TJMusiitwa** [comment]
@letsar Do you think this solution by @lukepighetti  is similar enough in practicality to the _flutter_sidekick_ **SidekickTeamBuilder**. 
Because that implementation is just what I would need for the project I am working on and seems _local_hero_ might be lacking something of the sort.

Maybe you could point to an example solution or advise on whether I should fork _flutter_sidekick_, converting to null_safety before using in my project?
Thanks

**Comment by MarazMia** [comment]
Can anyone give me one example code that works on grid?

**Comment by lukepighetti** [comment]
This might be helpful to you https://gist.github.com/lukepighetti/df460db180b9f6cb3410e3cc91ed74e6

**Comment by TDuffinNTU** [comment]
Bumping this, I'd love to see scrollable views working with this plugin! I wanted to use it for my notes app (which would use a grid cards to show previews of the saved notes) but couldn't get this to work since I'd need a scrolling view :(

I hope you come back to give this some love soon!

**Comment by abdelaziz-mahdy** [comment]
> Bumping this, I'd love to see scrollable views working with this plugin! I wanted to use it for my notes app (which would use a grid cards to show previews of the saved notes) but couldn't get this to work since I'd need a scrolling view :(
> 
> I hope you come back to give this some love soon!

Did you find any working solution, I need to animate grid changes too

**Comment by Snonky** [comment]
I took a shot at patching this in. Until @letsar finds time for the PR you can use my fork and maybe supply feedback if it achieves your needs.

To use the fork directly from git put this in your `pubspec.yaml`:

```
  local_hero:
    git:
      url: https://github.com/Snonky/local_hero.git
      ref: feature/only-remount
```

To enable the scroll support set `onlyAnimateRemount` to `true` in your `LocalHeroScope`.
This option supresses animations that come from position changes without tree surgery (e.g. scrolls or padding changes).

Before:
![local_hero_before](https://github.com/letsar/local_hero/assets/47639380/a5d75f66-bef7-4e17-b53c-ec2102a9fc7c)

After:
![local_hero_after](https://github.com/letsar/local_hero/assets/47639380/d12b46c7-87d1-47da-8f02-e196abf8e528)

**Comment by rafaeljustinox** [comment]
I would like to use it in `ListView` too.

Column | ListView
--|--
<video src="https://github.com/user-attachments/assets/98bb7c2d-c0f4-4e61-99ae-7daa80a3b9ce"></video> | <video src="https://github.com/user-attachments/assets/825c1793-55bd-4d1b-b434-6a95c309731c"></video>

**Comment by Snonky** [comment]
Can you post the minimum working code that created the lists from the videos?
